src/datreant/core/tests/test_treants.py

```diff
@@ -190,6 +190,50 @@ def test_remove_tags(self, treant):
             treant.tags.clear()
             assert len(treant.tags) == 0
 
+        def test_tags_set_behavior(self, tmpdir, treantclass):
+
+            with tmpdir.as_cwd():
+                # 1
+                t1 = treantclass('maple')
+                assert os.path.exists(t1.filepath)
+                t1.tags.add(['sprout', 'deciduous'])
+
+                # 2
+                t2 = treantclass('sequoia')
+                assert os.path.exists(t2.filepath)
+                t2.tags.add(['sprout', 'evergreen'])
+
+                tags_union = t1.tags | t2.tags
+                for t in ['sprout', 'deciduous', 'evergreen']:
+                    assert t in tags_union
+
+                tags_intersect = t1.tags & t2.tags
+                for t in ['sprout']:
+                    assert t in tags_intersect
```

Comment:
Add another loop here checking that 'deciduous' and 'evergreen' are *not* in `tags_intersect`.